server/plugin/utils_test.go

```diff
@@ -615,3 +617,101 @@ func TestGetConnectAccountFirstMessage(t *testing.T) {
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	p.API = mockAPI
+	for _, testCase := range []struct {
+		description       string
+		project           string
+		queryParamPage    string
+		queryParamPerPage string
+		expectedOffset    int
+		expectedLimit     int
+	}{
+		{
+			description:       "GetOffsetAndLimitFromQueryParams: valid page and per_page query params",
+			project:           "mockProject",
+			queryParamPage:    "1",
+			queryParamPerPage: "10",
+			expectedOffset:    10,
+			expectedLimit:     10,
+		},
+		{
+			description:       "GetOffsetAndLimitFromQueryParams: empty page and per_page query params",
+			project:           "mockProject",
+			queryParamPage:    "",
+			queryParamPerPage: "",
+			expectedOffset:    0,
+			expectedLimit:     constants.DefaultPerPageLimit,
+		},
+		{
+			description:       "GetOffsetAndLimitFromQueryParams: invalid page query param",
+			project:           "mockProject",
+			queryParamPage:    "invalidNonIntegerString",
+			queryParamPerPage: "10",
+			expectedOffset:    0,
+			expectedLimit:     10,
+		},
+		{
+			description:       "GetOffsetAndLimitFromQueryParams: invalid per_page query param",
+			project:           "mockProject",
+			queryParamPage:    "1",
+			queryParamPerPage: "invalidNonIntegerString",
+			expectedOffset:    constants.DefaultPerPageLimit,
+			expectedLimit:     constants.DefaultPerPageLimit,
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			mockAPI.On("LogError", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"))
+
+			req := httptest.NewRequest(http.MethodGet, fmt.Sprintf("%s/mockTeamID?project=%s&page=%s&per_page=%s", constants.PathGetSubscriptions, testCase.project, testCase.queryParamPage, testCase.queryParamPerPage), bytes.NewBufferString(`{}`))
+			req.Header.Add(constants.HeaderMattermostUserID, "mockMattermostUserID")
+
+			offset, limit := p.GetOffsetAndLimitFromQueryParams(req)
+
+			assert.Equal(t, testCase.expectedOffset, offset)
+			assert.Equal(t, testCase.expectedLimit, limit)
+		})
+	}
+}
+
+func TestParseSubscriptionsToCommandResponse(t *testing.T) {
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	p.API = mockAPI
+	for _, testCase := range []struct {
+		description       string
+		subscriptionsList []*serializers.SubscriptionDetails
+		channelID         string
+		createdBy         string
+		userID            string
+		command           string
+		message           string
+	}{
+		{
+			description:       "ParseSubscriptionsToCommandResponse: empty repos subscription list",
+			channelID:         "mockChannelID",
+			createdBy:         "mockUser",
+			userID:            "mockUserID",
+			command:           constants.CommandRepos,
+			subscriptionsList: []*serializers.SubscriptionDetails{},
+			message:           fmt.Sprintf("No %s subscription exists", constants.CommandRepos),
+		},
+		{
+			description:       "ParseSubscriptionsToCommandResponse: empty boards subscription list",
```

Comment:
don't we need to check cases when the repos or the boards list is not empty?

Reply:
yes we can check but up to here we are able to meet(74%) our target of 70% coverage so didn't include that

Reply:
If our coverage target is being met, we should not write unit tests for more functions, but if we are writing unit tests for a function, we should try to cover that function completely.